packages/components/lib/Header.js

```diff
@@ -0,0 +1,219 @@
+import React, { memo, isValidElement, cloneElement } from 'react'
```

Comment:
@PRASHANTK22111993  (optional) It would be great if things like steps and progress bar were there own standalone components that weren't dependent on the header. This way we could reuse them elsewhere in our UIs. If you think this is too much work for now, let's create a new card to separate them out in the future.

Reply:
Added the subtask and updated the PR and PR Descriptions.